firebase-database/src/androidMain/kotlin/dev/gitlive/firebase/database/database.kt

```diff
@@ -188,8 +189,30 @@ actual class DatabaseReference internal constructor(
     actual suspend fun removeValue() = android.removeValue()
         .run { if(persistenceEnabled) await() else awaitWhileOnline() }
         .run { Unit }
-}
 
+    actual suspend fun <T> runTransaction(strategy: KSerializer<T>, transactionUpdate: (currentData: T) -> T): DataSnapshot {
+        val deferred = CompletableDeferred<Result<DataSnapshot>>()
+        android.runTransaction(object : Transaction.Handler {
+
+            override fun doTransaction(currentData: MutableData) =
+                Transaction.success(transactionUpdate(decode(strategy, currentData)) as MutableData)
+
+            override fun onComplete(
+                error: DatabaseError?,
+                committed: Boolean,
+                snapshot: com.google.firebase.database.DataSnapshot?
+            ) {
+                if (error == null && snapshot != null) {
+                    deferred.complete(Result.success(DataSnapshot(snapshot)))
+                } else {
+                    deferred.complete(Result.failure(Throwable(error?.message)))
```

Comment:
we've got a database specific exception class you can throw here

Reply:
I dont think you need to use the Result type and use https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-completable-deferred/complete-exceptionally.html instead of throwing